tools/crm_resource_runtime.c

```diff
@@ -1713,3 +1752,87 @@ cli_resource_move(const char *rsc_id, const char *host_name, cib_t * cib, pe_wor
 
     return rc;
 }
+
+static void
+cli_resource_why_without_rsc_and_host(cib_t *cib_conn,GListPtr resources)
+{
+    GListPtr lpc = NULL;
+    GListPtr hosts = NULL;
+
+    for (lpc = resources; lpc != NULL; lpc = lpc->next) {
+        resource_t *rsc = (resource_t *) lpc->data;
+        rsc->fns->location(rsc, &hosts, TRUE);
+	if ( hosts == NULL ) {
+	    printf("Resource %s is not running",rsc->id);
+            cli_resource_check(cib_conn, rsc);
+	} else {
+	    printf("Resource %s is running",rsc->id);
+	}
+     }
+
+     g_list_free(hosts);
+     hosts = NULL;
```

Comment:
This free+null needs to be inside the for loop, otherwise the result from the previous iteration will affect the current one.